src/thunderbots/external/g3log.BUILD

```diff
@@ -0,0 +1,17 @@
+cc_library(
```

Comment:
Added a description for g3log, libusb, and munkres_cpp.

@henryymliu  - can you confirm what definitions you have in `generated_definitions.hpp` on osx? On ubuntu the file is basically empty.

Reply:
It's basically empty here too, other than 
`#define G3LOG_DEBUG DEBUG`

Reply:
...wondering how flexible Bazel is with inserting an header during the build, or having our own version of g3log (yikes?)

Reply:
There's been some discussion about this offline, summarizing here:
- we were able to run the cmake build in a `genrule`, so the header is no longer an issue here
- bazel defining `DEBUG` (it uses it for a logging level) is causing issues with this on macOs

At this point I think we should open a ticket for supporting MacOS and mark this comment as resolved to unblock this PR. @henryymliu can open the ticket and doc any issues you've seen that we'll need to address? Probably want to tag Akhil as well since I think he would be very interested in native MacOS support.

Reply:
Yep, opened #830 to track.